src/Stryker.Core/Stryker.Core.UnitTest/Baseline/Providers/AzureFileShareBaselineProviderTests.cs

```diff
@@ -1,476 +1,223 @@
-                  && req.RequestUri == expectedFileAllocationUri
-                  && req.Headers.Contains("x-ms-type")),
-                ItExpr.IsAny<CancellationToken>());
-
-            handlerMock
-              .Protected()
-              .Verify(
-               "SendAsync",
-               Times.Exactly(1),
-               ItExpr.Is<HttpRequestMessage>(req =>
-                  req.Method == HttpMethod.Put
-                  && req.RequestUri == expectedUploadContentUri),
-                ItExpr.IsAny<CancellationToken>());
+            // Arrange
+            var shareClient = Mock.Of<ShareClient>();
+            Mock.Get(shareClient).Setup(d => d.Exists(default)).Returns(Response.FromValue(true, default));
+            Mock.Get(shareClient).SetupGet(s => s.Uri).Returns(new Uri(_uri));
+
+            // root directory
+            var directoryClient = Mock.Of<ShareDirectoryClient>();
+            Mock.Get(shareClient)
+                .Setup(s => s.GetDirectoryClient("StrykerOutput"))
+                .Returns(directoryClient);
+            Mock.Get(directoryClient).Setup(d => d.Exists(default)).Returns(Response.FromValue(true, default));
+
+            // version directory
+            var subdirectoryClient = Mock.Of<ShareDirectoryClient>();
+            Mock.Get(directoryClient).Setup(d => d.GetSubdirectoryClient("v1")).Returns(subdirectoryClient);
+            Mock.Get(subdirectoryClient).Setup(d => d.Exists(default)).Returns(Response.FromValue(true, default));
+
+            // report file
+            var fileClient = Mock.Of<ShareFileClient>();
+            Mock.Get(subdirectoryClient).Setup(d => d.GetFileClient("stryker-report.json")).Returns(fileClient);
+            Mock.Get(fileClient).Setup(f => f.ExistsAsync(default)).Returns(Task.FromResult(Response.FromValue(true, default)));
+
+            // report file content download
+            var json = JsonReport.Build(new StrykerOptions(), ReportTestHelper.CreateProjectWith(), It.IsAny<TestProjectsInfo>()).ToJson();
+            var file = FilesModelFactory.StorageFileDownloadInfo(content: new MemoryStream(Encoding.Default.GetBytes(json)));
+            Mock.Get(fileClient).Setup(f => f.Download(null, default)).Returns(Response.FromValue(file, default));
+
+            // Act
+            var report = await new AzureFileShareBaselineProvider(new StrykerOptions(), shareClient).Load("v1");
+
+            // Assert
+            report.ShouldNotBeNull();
+
+            Mock.Get(shareClient).VerifyAll();
+            Mock.Get(shareClient).VerifyNoOtherCalls();
+
+            Mock.Get(directoryClient).VerifyAll();
+            Mock.Get(directoryClient).VerifyNoOtherCalls();
+
+            Mock.Get(subdirectoryClient).VerifyAll();
+            Mock.Get(subdirectoryClient).VerifyNoOtherCalls();
+
+            Mock.Get(fileClient).VerifyAll();
+            Mock.Get(fileClient).VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task Save_Report()
```

Comment:
Does this mimic the upload of a large or small file? I think we need tests for both.